### DISCUSSION TITLE
Problem connecting on LAN (WAN works flawlessly)

### DISCUSSION THREAD

**Discussion by quietsnow** [homeassistant-apps/app-cloudflared#461] [Q&A]
If I am not on my local network, my tunnel works flawlessly, every time. Kudos to @brenner-tobias. Easiest cloudflare tunnel to setup! However, I'd love to be able to use the same link when I'm at home (instead of the LAN IP). I have tried so many different things and none of them have fixed this issue. Short of adding a DNS entry on my home router,  is there another way to connect to the WAN URL locally? 

Another cloudflare tunnel I have works fine from the LAN and WAN. Zero Trust configurations are very similar (the only difference is that HA goes to http://homeassistant:port; and the other one goes to http://localhost:port). Thank you.

**Comment by brenner-tobias** [comment]
As far as I am aware, without altering local DNS records (either in your device or in your router), there is no possibility to do that for a PC (in the mobile App, you can configure a local IP and a WAN IP, if you are not in your home wifi).
The reason is that your HA interface is served from the Cloudflare Reverse Proxy Server from the perspective of the client. So there is no way to "bypass" this, if you tell it to use the domain. 
For me, I am simply using my domain at home as well. This should work flawlessly, the downside being that the location is router via the internet and not locally. At the same time, I do not think it is too much of an inconvenience to simply have another bookmark in your browser that you use when being at home.

**Comment by quietsnow** [comment]
You are correct. Turns out that I already had the DNS entries in the router. The gremlin was elsewhere. Thanks for your help. All working. :)
